qiskit_ibm_runtime/transpiler/passes/basis/fold_rzz_angle.py

```diff
@@ -0,0 +1,185 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Pass to wrap Rzz gate angle in calibrated range of 0-pi/2."""
+
+from typing import Tuple
+from math import pi
+
+from qiskit.circuit.library.standard_gates import RZZGate, RZGate, XGate
+from qiskit.circuit.parameterexpression import ParameterExpression
+from qiskit.circuit import Qubit
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+
+import numpy as np
+
+
+class FoldRzzAngle(TransformationPass):
+    """Fold Rzz gate angle into calibrated range of 0-pi/2 with
+    local gate tweaks.
+
+    This pass preserves the number of Rzz gates, but it may add
+    extra single qubit gate operations.
+    These gates might be reduced by the following
+    single qubit optimization passes.
+    """
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        # Currently control flow ops and Rzz cannot live in the same circuit.
+        # Once it's supported, we need to recursively check subroutines.
```

Comment:
I prefer to already support subroutines, to avoid pitfalls when the time comes and dynamic circuits are allowed.

Reply:
I usually don't like writing unreachable logic because of the maintenance overhead. But the code is in [013a6b4](https://github.com/Qiskit/qiskit-ibm-runtime/pull/2043/commits/013a6b45f54138587ee95ec7ead0e31250d58570)